frontend2/src/views/TeamProfile.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useMemo } from "react";
+import { useEpisodeId } from "contexts/EpisodeContext";
+import { useParams } from "react-router-dom";
+import { useQueryClient } from "@tanstack/react-query";
+import { useTeam } from "api/team/useTeam";
+import { isNil } from "lodash";
+import SectionCard from "components/SectionCard";
+import { PageTitle } from "components/elements/BattlecodeStyle";
+import MemberList from "components/team/MemberList";
+import Loading from "components/Loading";
+import { Status526Enum } from "api/_autogen";
+import { useEpisodeInfo } from "api/episode/useEpisode";
+
+const isNilOrEmptyStr = (str: string | undefined | null): boolean => {
+  return isNil(str) || str === "";
+};
+
+// rendered at /:episodeId/team/:teamId
+const TeamProfile: React.FC = () => {
+  const { episodeId } = useEpisodeId();
+  const { teamId } = useParams();
+  const episode = useEpisodeInfo({ id: episodeId });
+  const team = useTeam({ episodeId, id: teamId ?? "" });
+  const membersList = useMemo(() => {
+    return (
+      <div className="flex flex-col gap-8">
+        {!team.isSuccess ? (
+          <Loading />
+        ) : (
+          <MemberList members={team.data.members} />
+        )}
+      </div>
+    );
+  }, [team]);
+  const eligibles = useMemo(
+    () =>
+      episode.data?.eligibility_criteria.filter(
+        (criterion) =>
+          team.data?.profile?.eligible_for?.find(
+            (id) => id === criterion.id,
+          ) !== undefined,
+      ) ?? [],
+    [episode, team],
+  );
+  if (isNil(team.data)) {
```

Comment:
Should this be `if isNil && !isLoading`? Or does this successfully still load until data is not returned?